linera-client/src/benchmark.rs

```diff
@@ -162,6 +217,68 @@ where
             });
         }
 
+        let metrics_watcher = if let Some(health_check_endpoints) = health_check_endpoints {
+            let metrics_addresses = health_check_endpoints
+                .split(',')
+                .map(|address| address.trim())
+                .map(|address| {
+                    let parts = address.split(':').collect::<Vec<_>>();
+                    ensure!(
+                        parts.len() == 2,
+                        BenchmarkError::ParseValidatorMetricsAddress(address.to_string())
+                    );
+                    let host = parts[0].to_owned();
+                    let port = parts[1]
+                        .parse::<u16>()
+                        .map_err(BenchmarkError::ParseValidatorMetricsPort)?;
+                    Ok(format!("http://{}:{}/metrics", host, port))
```

Comment:
Do we really gain much by this partial validation of the address format? The part before the colon could still be invalid, after all.

```suggestion
                    format!("http://{}/metrics", address)
```